src/components/UI/market/ProductListItem.tsx

```diff
@@ -1,14 +1,25 @@
 import styled from 'styled-components';
-import { product } from '../../../pages/Market';
+import { useContext } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { ProductContext } from '../../../ProductContext';
+import { Iproduct } from '../../../pages/Market';
 import heart_empty from '../../../assets/heart-empty.svg';
 
 interface ProductListItemProps {
-  product: product;
+  product: Iproduct;
 }
 
 const ProductListItem: React.FC<ProductListItemProps> = ({ product }) => {
+  const productContext = useContext(ProductContext);
+  const navigate = useNavigate();
+
+  const clickItem = () => {
+    navigate(`/${product.productId}`);
+    productContext.setProductId(product.productId);
+  };
+
   return (
-    <ItemBox>
+    <ItemBox onClick={() => clickItem()}>
```

Comment:
1. 아래 정도로 줄여서 쓸 수도 있습니다
```suggestion
    <ItemBox onClick={clickItem}>
```

2. onClick handler 함수의 네이밍은 on\*Click 혹은 handle\*Click 의 형태를 많이 사용해요. (ex. onItemClick) 조금 오래된 글이지만 참고해보시면 좋을듯합니다. [참고](https://jaketrent.com/post/naming-event-handlers-react/)

Reply:
수정했습니다!!!